sys/posix/pthread/include/pthread_once.h

```diff
@@ -24,14 +24,17 @@ extern "C" {
 /**
  * @brief           Datatype to supply to pthread_once().
  */
-typedef volatile int pthread_once_t;
+typedef struct {
+  int   is_initialized;  /**< initialized */
+  int   init_executed;   /**< init function executed */
+} pthread_once_t;
```

Comment:
Maybe add a comment that this struct needs to be in sync with the one in newlib

Reply:
Will do, may I squash it directly?

Reply:
Sure!